packages/steiger-plugin-fsd/src/_lib/index-source-files.ts

```diff
@@ -28,6 +28,13 @@ export function indexSourceFiles(root: Folder): Record<string, SourceFile> {
   }
 
   for (const [layerName, layer] of Object.entries(getLayers(root))) {
+    // Even though files that are directly inside a layer are not encouraged by the FSD and are forbidden in most cases
+    // (except for an index/root file for the app layer as an entry point to the application), users can still add them.
+    // So, we need to index all files directly inside a layer to find errors.
+    layer.children
+      .filter((child) => child.type === 'file')
+      .forEach((file) => walk(file, { layerName: layerName as LayerName, sliceName: null, segmentName: null }))
```

Comment:
suggestion: could you add a test for this case please? The tests are in the same file, in the `if (import.meta.vitest)` thing

Reply:
Yes, done